File: jobportal-backend/src/main/java/com/kiwisoft/jobportal/service/PartnerService.java
package com.kiwisoft.jobportal.service;

import com.kiwisoft.jobportal.dto.request.PartnerRequest;
import com.kiwisoft.jobportal.dto.response.PartnerResponse;
import com.kiwisoft.jobportal.dto.request.UpdatePartnerRequest;
import java.util.List;
import org.springframework.data.domain.Page;
public interface PartnerService {

    PartnerResponse createPartner(PartnerRequest request);

    List<PartnerResponse> getAllPartners();

    PartnerResponse getPartnerById(Long id);

    PartnerResponse updatePartner(
            Long id,
            UpdatePartnerRequest request
    );

    PartnerResponse blockPartner(Long id);

    PartnerResponse deletePartner(Long id);

    Page<PartnerResponse> getPartnersPage(
            int page,
            int size
    );

    Page<PartnerResponse> searchPartnersPage(
            String keyword,
            int page,
            int size
    );
}

File: jobportal-frontend/src/services/partnerService.js
import api from "./api";

// Get All Partners
export const getAllPartners = async () => {
  const res = await api.get("/partners");
  return res.data;
};

// Get Partner By ID
export const getPartnerById = async (id) => {
  const res = await api.get(`/partners/${id}`);
  return res.data;
};

// Create Partner
export const createPartner = async (partner) => {
  const res = await api.post("/partners", partner);
  return res.data;
};

// Update Partner
export const updatePartner = async (id, partner) => {
  const res = await api.put(`/partners/${id}`, partner);
  return res.data;
};

// Delete Partner
export const deletePartner = async (id) => {
  const res = await api.delete(`/partners/${id}`);
  return res.data;
};

// Block Partner
export const blockPartner = async (id) => {
  const res = await api.post(`/partners/${id}/block`);
  return res.data;
};